### PR TITLE
Add input format in Transpose GetHash

### PR DESCRIPTION
**Pull request by lidanqing-vv**
Add input format in Transpose GetHash, resolving transpose quantization bug. With this fix, mobilenet-ssd INT8 tester will work.

### CODE REVIEW

**Review by wojtuss — APPROVED** [review]

File: paddle/fluid/operators/mkldnn/transpose_mkldnn_op.cc [hunk 1]
```diff
@@ -124,8 +124,7 @@ class TransposeMKLDNNGradOpKernel : public paddle::framework::OpKernel<T> {
         paddle::framework::vectorize2int(out_grad->dims());
 
     const std::string key = platform::TransposeMKLDNNHandler::GetHash(
-        nchw_tz, axis, ctx.op().Output(framework::GradVarName("X")) +
-                           std::to_string(out_grad->format()));
+        nchw_tz, axis, ctx.op().Output(framework::GradVarName("X")));
```
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/transpose_mkldnn_op.cc]: Why remove the `out_grad->format()` in `TransposeMKLDNNGradOpKernel`?

**Comment by lidanqing-vv** [comment]
Without adding input->format() into hash, INT8 `warmup stage` and `inference stage` may have same hashkey for `transpose` operator.

For example, when batch_size=warm_up_size=50, 
warm_up stage: `Transpose:    in:f32_nChw16c out:f32_blocked,num:1,50x12x19x19`
inference stage: `Transpose:     in:s8_nhwc out:f32_nchw,num:1,50x12x19x19`

File: paddle/fluid/operators/mkldnn/transpose_mkldnn_op.cc (hunk 1 above)
Comment (lidanqing-vv) [reply paddle/fluid/operators/mkldnn/transpose_mkldnn_op.cc]: @luotao1  Hi! Because in INT8 inference, we don't use `TransposeMKLDNNGradOpKernel`(which is for backpropogation and training). @wojtuss suggests that we should not influence others' design. About why we add `input->format()`, please check following comment.

Comment (luotao1) [reply paddle/fluid/operators/mkldnn/transpose_mkldnn_op.cc]: Got it!
